### DISCUSSION TITLE
Allow Custom Search Engine to use file:/// addresses

### DISCUSSION THREAD

**Discussion by SoraHjort** [Floorp-Projects/Floorp#780] [Feature Request]
This is a bit of an odd and niche request, so I won't blame anyone if it gets rejected.

`file:///` addresses currently are not allowed to be used when adding a custom search engine. Attempts to do so just results in no search engine creation. I would like to request a change to make it allowed.

Usage reason: I use a local html file, with some javascript on it, to act as a search query parser. To primarily prevent accidental searches, certain browsers have made me paranoid because they don't autocomplete quick enough. While also letting me quickly select which search site I would like to parse it through. (Hosted a functioning demo on github pages of the [html page](https://sorahjort.github.io/search.html#floorp%20browser%20-"floor") for the curious)

Of course, for all I know, there is some security concern to prevent `file:///` from being allowed. As well as I would probably be one of the very few, if any, that would use it. But it is also something that I know chromium browsers allows you to do when creating a custom search, if that makes any difference.

Anyway, I'm liking the browser so far. It feels nice to be able to go back to using the gecko browser engine.

**Comment by Telmesomething** [comment]
You could try using the free Everything Search from https://www.voidtools.com/ and then use "html" as a search string. Right click the "Home" header column to define your additional search options.  

I use it as a general search tool rather than using the default OS search utility. There's a review of it over at [Ghacks.net](https://www.ghacks.net/2020/10/09/add-everything-search-to-the-windows-taskbar-for-even-faster-searches/) which is where I first discovered it myself.

**Reply by SoraHjort** [reply]
I... what? What does a file search program that is not accessible by the browser have to do with setting a custom search engine? I'm quite confused by this recommendation.

**Reply by Telmesomething** [reply]
I was suggesting using a third party tool to achieve what you're looking for since it's not possible to develop a search tool for a file which resides solely on your machine.

**Reply by SoraHjort** [reply]
> ...develop a search tool for a file...

Ok, I see where you're getting confused. No I'm not looking for the browser to search files on the computer. I'm not quite sure why you got that, but let me explain it another way.

I just want to be able to set the URL address in the custom search engine to a local file. Setting the custom search engine to something like `file:///C:/Path/To/File.html#%s`

The issue is that when you try to do so, like so: 

![image](https://github.com/Floorp-Projects/Floorp/assets/18604504/76bb4ee9-821d-42e4-8a50-d21cfb4c6529)

The moment you click "Add engine" it doesn't create the entry in the search engine list. Thus, `file:///` would need to be added to the list of acceptable search engine paths, much like how `http://` and `https://` are. Other browsers that let you set custom search engines do let you set ones with `file:///`, so it would be nice to let Floorp to do the same.

**Reply by Telmesomething** [reply]
In your screenshot for Path, replace the "#" character with the equals character "="

**Reply by SoraHjort** [reply]
That doesn't make it work. The issue is that `file:///` needs to be whitelisted. It isn't the `#` or the lack of `=`.
You could add any url to the search and it'd work as long as it's `http://` or `https://`

Such as but very not limited to: 
- `http://fakeurl.fakewebsite`
- `http://fakeurl.fakewebsite#%s`
- `https://fakeurl.fakewebsite#%s`

Those three examples can be added via the custom search with no issue. They're not valid sites, but they do show you can use `#` and still be considered valid since they will show up in the list upon clicking Add Engine.

The issue is strictly with the `file:///` protocol not being allowed for some reason.
